lib/travis/api/v3/queries/log.rb

```diff
@@ -0,0 +1,74 @@
+module Travis::API::V3
+  class Queries::Log < Query
+    require 'net/http'
+    require 'uri'
+    require 'aws/s3'
+
+    FORMAT = "Log removed by %s at %s"
+
+    def find(job)
+      #check for the log in the Logs DB
+      log = Models::Log.find_by_job_id(job.id)
+      raise EntityMissing, 'log not found'.freeze if log.nil?
+      #if log has been aggregated, look at log.content
+      if log.aggregated_at
+        create_log_parts(log, log.content)
+      #if the log has been archived, go to s3
+      elsif log.archived_at
+        archived_log_path = archive_url("/jobs/#{job.id}/log.txt")
+        content = Net::HTTP.get(URI.parse(archived_log_path))
+        create_log_parts(log, content)
+      end
+      log
+    end
+
+    def create_log_parts(log, content)
+      log_part = Models::LogPart.new(log_id: log.id, content: content, number: 0, created_at: log.created_at)
```

Comment:
I believe AR will set the relation to [] if there aren't any yet so could this be:
``` ruby
log.log_parts << Models::LogPart.new(log_id: log.id, content: content, number: 0, created_at: log.created_at)
```
or does log.log_parts already have something there sometimes we want to override?